spot_driver/spot_driver/spot_ros2.py

```diff
@@ -844,6 +844,31 @@ def __init__(self, parameter_list: Optional[typing.List[Parameter]] = None, **kw
                 if self.auto_stand.value:
                     self.spot_wrapper.stand()
 
+        # Explicitly get a lease when get_lease_on_action is False
+        # because spot_wrapper does not get a lease at initialization.
+        if self.spot_wrapper is not None and not self.get_lease_on_action.value:
+            self.spot_wrapper.getLease()
+        self.create_service(
+            srv_type=Trigger,
+            srv_name="gen_new_spot_ros_lease",
+            callback=self.new_lease_callback,
+            callback_group=self.group,
+        )
+
+    def new_lease_callback(self, request: Trigger.Request, response: Trigger.Response) -> Trigger.Response:
+        self.get_logger().info("Incoming request to generate a new lease.")
+        if self.spot_wrapper is None:
+            response.success = True
+            response.message = "SpotROS is running in mock mode."
+            return response
+
+        old_lease = self.spot_wrapper._lease
+        self.spot_wrapper.getLease()
```

Comment:
Should we update `getLease` to return the lease? It wouldn't break existing code and then you wouldn't need to access a private member here

Reply:
I think that could be a good idea, but also raises potential confusion that this is just a getter instead of an "action" in the BD API.

Reply:
I support having `getLease` return the lease. I didn't do it originally because I thought I'd make minimal changes in the first go. 

I think the right way to resolve the confusion is to rename it, to something like `genLease`.

Reply:
I am in favor of reducing confusion, and I think the right way to do that is to avoid introducing terminology. The BD API terminology regarding this is "take" and "acquire", and wrapping those two terms as "get" or "generate" or any other name seems to be unnecessary confusion.

I imagine there is a convenience reason to have both "take" and "acquire" wrapped under one name, and the chosen option depending on `SpotWrapper` initialization:

```
            use_take_lease: Use take instead of acquire to get leases. This will forcefully take the lease from any other lease owner.
```.

Can we just call it "takeOrAcquireLease"?

Reply:
> I am in favor of reducing confusion, and I think the right way to do that is to avoid introducing terminology. The BD API terminology regarding this is "take" and "acquire", and wrapping those two terms as "get" or "generate" or any other name seems to be unnecessary confusion.
> 
> I imagine there is a convenience reason to have both "take" and "acquire" wrapped under one name, and the chosen option depending on `SpotWrapper` initialization:
> 
> ```
>             use_take_lease: Use take instead of acquire to get leases. This will forcefully take the lease from any other lease owner.
> ```
> 
> @amessing-bdai Can we just call it `takeOrAcquireLease`, and deprecate `getLease`?

We can update the function name, but would also need to check/PR  spot_ros (as in ros 1).

Reply:
I put up https://github.com/bdaiinstitute/spot_wrapper/pull/83https://github.com/bdaiinstitute/spot_wrapper/pull/83 which exposes `_lease` as `lease2` (`lease` was taken. better name suggestions are welcome)

We can worry about updating the function name and deprecating the old uses separately.